Mathlib/Data/Finset/Image.lean

```diff
@@ -656,6 +657,26 @@ theorem range_sdiff_zero {n : ℕ} : range (n + 1) \ {0} = (range n).image Nat.s
   rw [range_succ, image_insert, ← hk, insert_sdiff_of_not_mem]
   simp
 
+variable {ι : Type*} [Preorder α] {s : Finset α}
+
+lemma exists_maximalFor (f : ι → α) (s : Finset ι) (hs : s.Nonempty) :
+    ∃ i, MaximalFor (· ∈ s) f i := by
```

Comment:
What's the reasoning for putting these lemmas in this specific file?

Reply:
No better file, but I don't like it. What do you think of creating `Order.Preorder.Finite` and including results like this there? Maybe `Data.Fintype.Order` could similarly become `Order.CompleteLattice.Finite`

Reply:
I think there are lots of finiteness / minimality results scattered across different files, which makes them all really annoying to find. So if they could be grouped in some way, then that seems the best option